client/http-client/src/client.rs

```diff
@@ -418,8 +418,8 @@ where
 		let id_range = generate_batch_id_range(id, batch.len() as u64)?;
 
 		let mut batch_request = Vec::with_capacity(batch.len());
-		for ((method, params), id) in batch.into_iter().zip(id_range.clone()) {
-			let id = self.id_manager.as_id_kind().into_id(id);
+		for ((method, params), _id) in batch.into_iter().zip(id_range.clone()) {
```

Comment:
revert this, I don't see reason to change this.

for batches we are generating (next_id..next_id+batch_len)

Reply:
I am working on some improvements on the design to make batching possible again

Reply:
Ok, sorry about that I didn't realize it until I saw the code again

Reply:
No problem at all. I made it a member function of `RequestId`. Let me know what you think